modules/javascript/web-worker/load.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * Module Name: Web Worker
+ * Description: Run JavaScript in a separate Web Worker with the help of Partytown library.
+ * Experimental: Yes
+ *
+ * @since   1.0.0
+ * @package performance-lab
+ */
+
+/**
+ * PartyTown Configuration
+ *
+ * @since 1.0.0
+ * @see https://partytown.builder.io/configuration
+ * @return void
+ */
+function web_worker_partytown_configuration() {
+	$config = array(
+		'lib' => str_replace( site_url(), '', plugin_dir_url( __FILE__ ) ) . 'assets/js/partytown/',
+	);
+
+	/**
+	 * Add configuration for PartyTown.
+	 *
+	 * @since 1.0.0
+	 * @param array $config Configuration for PartyTown.
+	 * @return array
+	 */
+	$config = apply_filters( 'partytown_configuration', $config );
+
+	?>
+	<script>
+		window.partytown = <?php echo wp_json_encode( $config ); ?>;
+	</script>
+	<?php
+}
+add_action( 'wp_head', 'web_worker_partytown_configuration', 1 );
+
+/**
+ * Initialize PartyTown
+ *
+ * @since 1.0.0
+ * @return void
+ */
+function web_worker_partytown_init() {
+	wp_enqueue_script(
```

Comment:
I think we should only register the script at this stage and only enqueue in the scenarios where is required instead of loading it all the time. Additionally is this required to be loaded at the top of the page instead of in the `<footer>` ?
```suggestion
	wp_register_script(
```

Reply:
> I think we should only register the script at this stage and only enqueue in the scenarios where is required instead of loading it all the time.

Partytown script should be enqueued before any other script in order to queue third-party script in a Worker thread so IMO it should be enqueued instead. 

> Additionally is this required to be loaded at the top of the page instead of in the <footer> ?

It should be loaded in the head as most of the time third-party scripts like Google Analytics are enqueued in the head.

Reply:
> Additionally is this required to be loaded at the top of the page instead of in the `<footer>`?

Ideally, yes, the script should be loaded in the `<head>`, not in the `<footer>`.

> The partytown.js file must be in the `<head>` before all other third-party scripts, and should not have the async or defer attributes. Because the script must be executed immediately, and since it’s such a small file, it’s always preferred to inline the partytown.js script content, rather than it being an additional http request.

https://partytown.builder.io/distribution

> I think we should only register the script at this stage and only enqueue in the scenarios where is required instead of loading it all the time.

Yes, that'd be better.

Reply:
> Also, a check can be added to check if some script's dependency is partytown then only enqueue the Partytwon script

I have a second thought about this: what happens if people add `scripts` manually without using `wp_enqueue_script`? [Something like this example](https://github.com/rtCamp/wp-partytown/blob/develop/examples/post-request.php#L21-L56). Is it something the plugin should support?

Reply:
@luisherranz You mentioned the nice use case. In this case, I think we need to enqueue it each time.

Reply:
> It should be loaded in the head as most of the time third-party scripts like Google Analytics are enqueued in the head.

Cool, this inline would impact the TTFB by having this inline executed at the top of the page.

My point by using a `wp_register_script` is the fact that we can deregister from an external plugin in the case where we don't want to execute Partytown so the plugin effectively first registers the script and then enqueue happens, by doing this as well it allows for other scripts that depend on this script to be enqueued when dependencies from party town are enqueued. 

Additionally this mentions that this should work every time a page loads similarly to Google Analytics, maybe I'm not understanding the use case, would you mind expanding a bit so I can have a better sense in that regard.

Reply:
The main partytown script needs to be a blocking `<head>` script. Not inlining it means that the browser needs to do an additional blocking round-trip to the server, which would delay the page load much more than adding those extra kbs to the HTML.

Reply:
> would you mind expanding a bit so I can have a better sense in that regard

There are [some articles in their docs](https://partytown.builder.io/#articles) that you may find interesting, especially [the first one](https://dev.to/adamdbradley/introducing-partytown-run-third-party-scripts-from-a-web-worker-2cnp). If you want to see it in action, this [video with Adam Bradley](https://www.youtube.com/watch?v=btr1CiHMvzU) is a good resource.

Reply:
inlining is fine, however:
* `partytown` should only be added if used on the page (eg. some other script depends on it)

> the fact that we can deregister from an external plugin

I don't think this is an important concern for the plugin implementation (the module can already be disabled programmatically) , especially since plugins have to opt in so far.